### PR TITLE
fop/fom.c:readyit(): check that the fom is on the wait list

### PR DESCRIPTION
**Pull request by nikitadanilov**
It is possible that by the time readyit() call-back, which was posted by
m0_fop_wakeup(), is invoked, the fom has already been woken up, or has already
terminated.

Check that the fom is on the wait list before doing anything else.

Signed-off-by: Nikita Danilov <nikita.danilov@seagate.com>

### CODE REVIEW

**Review by andriytk — APPROVED** [review]
The same change was proposed at https://github.com/Seagate/cortx-motr/pull/1471.

File: fop/fom.c [hunk 1]
```diff
@@ -438,11 +438,25 @@ M0_INTERNAL void m0_fom_ready(struct m0_fom *fom)
...
+	struct m0_fom_locality *floc = M0_AMB(floc, grp, fl_group);
+	/*
+	 * Careful, do not touch fom fields before checking it is on the wait
+	 * list: it is possible that the fom is already on the run queue, or has
+	 * terminated.
+	 *
+	 * @todo Scanning the entire (potentially very long) wait list on each
+	 * wakeup is not scalable.
+	 */
+	if (wail_tlist_contains(&floc->fl_wail, fom))
```
Comment (huanghua78) [inline fop/fom.c:458]: In #1471, @andriytk proposed :
```
if (m0_fom_is_waiting(fom))
        m0_fom_ready(fom);
```

How do you think, @nikitadanilov ?

Comment (huanghua78) [reply fop/fom.c:458]: It's fine here.

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**